### PR TITLE
level aa contrast compliance faa other income

### PR DESCRIPTION
**Pull request by ATBull81**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [ ] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/186499199

# A brief description of the changes

Current behavior: Keyboard navigation for cancel and save buttons are lacking, edit and delete buttons are inaccessible, no fallback text for checkmarks or edit/delete buttons

New behavior: All behaviors improved, navigation improved.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name: CONTRAST_LEVEL_AA_IS_ENABLED

- [ ] DC
- [x] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

Screenshot 2024-01-03 at 1 16 09 PM
Screenshot 2024-01-03 at 12 26 44 PM

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

**Review by mdkaraman — DISMISSED** [review]
Nice work!

File: components/financial_assistance/app/views/financial_assistance/shared/_fa_left_nav.html.erb [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
         <li class="list-box step-tabs <%= li_nav_classes_for(:tax_info) %> applicant-nav">
           <% if @applicant.tax_info_complete?%>
-            <span class="fa fa-check applicant-check"></span>
+            <span class="fa fa-check applicant-check"><span class="hide"><%=l10n(".completed")%></span></span>
```
Comment (scaustin34) [inline components/financial_assistance/app/views/financial_assistance/shared/_fa_left_nav.html.erb:16]: I could be wrong, but i don't think the l10n is actually referencing the value in the translation file as is. I think the . is incorrect syntax? 

We have some processing on l10n fields that, as a fallback if it cant find it in the translation file, will take whatever is in the quotes and just add spaces and correct case. I think is actually why this is probably rendering correctly without being actually linked.

**Review by scaustin34 — CHANGES_REQUESTED** [review]

File: components/financial_assistance/app/views/financial_assistance/shared/_fa_left_nav.html.erb (hunk 1 above)
Comment (ATBull81) [reply components/financial_assistance/app/views/financial_assistance/shared/_fa_left_nav.html.erb]: After some discussion off the ticket and doing some testing with it, `l10n("completed")` is the proper format (unless it is stored in the translations file as `"en.insured.completed"`, which would need us to use `l10n("insured.completed")`). I have updated this ticket and tried to fix places I have used the incorrect format previously: https://github.com/ideacrew/enroll/pull/3431/commits/c205c244b2dd738a90f1e9fa653c7218e315adb0

**Review by mdkaraman — DISMISSED** [review]
Uses correct syntax for translation helper calls.  Good catch @scaustin34!

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js [hunk 2]
```diff
@@ -386,10 +388,20 @@ document.addEventListener("turbolinks:load", function () {
       var clonedForm = newIncomeForm.clone(true, true)
         .removeClass('hidden')
         .appendTo(incomeListEl);
+      if (incomeListEl.children().length > 1 && incomeListEl.children().first().attr('id') === 'hidden-income-form') {
+        console.log("removed by otherindex check??");
+        incomeListEl.children().first().remove();
+      }
+      if (unemploymentIndex != 0) {
+        var previousForm = clonedForm.prev('.new-unemployment-income-form');
+        console.log("removed by otherindex check??");
```
Comment (scaustin34) [inline components/financial_assistance/app/assets/javascripts/financial_assistance/income.js:397]: probably should pull this little console log guy out, if you have time

**Review by scaustin34 — APPROVED** [review]

**Review by scaustin34 — DISMISSED** [review]

File: components/financial_assistance/app/assets/javascripts/financial_assistance/income.js (hunk 2 above)
Comment (ATBull81) [reply components/financial_assistance/app/assets/javascripts/financial_assistance/income.js]: removed thanks https://github.com/ideacrew/enroll/pull/3431/commits/4430229198d3828e7d0d7ed520925f7726e64e2b

**Review by mdkaraman — APPROVED** [review]

**Review by scaustin34 — APPROVED** [review]

**Pull request merged**